### PR TITLE
【BIT】引擎补齐：functional.grid_sample functional.hsigmoid_loss functional.upsample Tensor.log_normal_  Tensor.index_select

### PR DESCRIPTION
**Pull request by mzj104**
### **修改内容**
------------------

 **主要工作**

1. 修改5个api

 **具体api修改**

1. **functional.grid_sample**
**修改依据**：错误原因是不支持float16的数据类型
**改正方法**：在get_numpy_tensor()函数中，将数据类型改为float32

2. **functional.hsigmoid_loss**
**修改依据**：错误原因有3：
>  label.max() < num_classes，否则报错或 crash。
>  path_table[i][j] 必须 < weight.shape[0]，否则底层越界崩溃。
>  path_code 中只能是 0 或 1。
**改正方法**：在get_numpy_tensor()函数中，添加随机数范围限制
其中下面这个配置最后一个参数为true，涉及稀疏处理，会报错，暂时不好修复，放在了fresh_report
`paddle.nn.functional.hsigmoid_loss(Tensor([4, 8],"float64"), Tensor([4],"int64"), 6, Tensor([5, 8],"float64"), Tensor([5, 1],"float64"), Tensor([4, 5],"int64"), Tensor([4, 5],"int64"), True, )`

3. **functional.upsample**
**修改依据**：错误原因是size或scale_factor没有正确初始化，size需要保证大于0的整数；scale_factor需要是正数，且大于0.5（否则输入维度等于1时，缩放后的维度小于1，会报错）
**改正方法**：因为size和scale_factor不会同时出现非None的值，因此检测到size非None就随机size，检测到scale_factor非None就随机scale_factor

4. **Tensor.log_normal_**
**修改依据**：不用修改，直接就能通过

5. **Tensor.index_select**
**修改依据**：错误原因是index没有正确初始化
**改正方法**：检测是否含有axis参数，若没有则设为0；index应该小于inputs[axis]

 **运行结果**
配置均pass
image

另外：
之前的pr里面把api_config_merged_not_support_tensor_init.txt里面修复完成的配置放在了api_config_merged_9.txt，现在把它们调整了回来，放在了api_config_merged_tensor_init.txt

之前pr中关于expand的放在fresh_report的特殊情况，其实不应该是特殊情况，是能够处理的，使用广播机制即可，现已修改。

### CODE REVIEW

File: report/fresh_report/test_log_expand [hunk 1]
```diff
@@ -1 +1,8 @@
-paddle.expand(x=Tensor([2, 2],"int64"), shape=Tensor([3],"int32"), )
\ No newline at end of file
+paddle.expand(x=Tensor([2, 2],"int64"), shape=Tensor([3],"int32"), )
+aistudio@jupyter-3165291-8843119:~/PaddleAPITest$ python engine.py --paddle_only=True --api_config='paddle.expand(x=Tensor([2, 2],"int64"), shape=Tensor([3],"int32"), )'
+test begin: paddle.expand(x=Tensor([2, 2],"int64"), shape=Tensor([3],"int32"), )
+W0408 07:29:09.126283 46182 gpu_resources.cc:119] Please NOTE: device: 0, GPU Compute Capability: 7.0, Driver API Version: 12.0, Runtime API Version: 11.8
+W0408 07:29:09.127291 46182 gpu_resources.cc:164] device: 0, cuDNN Version: 8.9.
+[paddle error] paddle.expand(x=Tensor([2, 2],"int64"), shape=Tensor([3],"int32"), ) 
+ (InvalidArgument) The number of elements (1) of 'shape' for expand_v2 op must be greater than or equal to the rank (2) of the input.
+  [Hint: Expected expand_shape.size() >= static_cast<size_t>(x_dims.size()), but received expand_shape.size():1 < static_cast<size_t>(x_dims.size()):2.] (at ../paddle/phi/infermeta/unary.cc:1311)
```
Comment (wanghuancoder) [inline report/fresh_report/test_log_expand:8]: 这个看起来是shape=Tensor([3],"int32")生成的值不符合要求导致的啊？

Comment (mzj104) [reply report/fresh_report/test_log_expand]: 之前pr中关于expand的放在fresh_report的特殊情况，其实不应该是特殊情况，是能够处理的，使用广播机制即可，现已修改。

**Review by wanghuancoder — APPROVED** [review]
LGTM
